salt/states/sysctl.py

```diff
@@ -52,7 +52,7 @@ def present(name, value, config=None):
             config = "/etc/sysctl.conf"
 
     if __opts__["test"]:
-        current = __salt__["sysctl.show"]()
+        current = {name: __salt__["sysctl.get"](name)}
```

Comment:
Adding name to this dictionary will lead to this line `if name in current and name not in configured:` always passing the first condition, and we only want to pass that condition if that name actually exists.

Reply:
Fixed